src/http/index.js

```diff
@@ -89,24 +107,28 @@ class HttpApi {
 
     const config = await ipfs.config.get()
 
-    const apiAddr = config.Addresses.API.split('/')
-    const apiServer = await this._createApiServer(apiAddr[2], apiAddr[4], ipfs)
-    await apiServer.start()
-    apiServer.info.ma = hapiInfoToMultiaddr(apiServer.info)
-    this._apiServer = apiServer
+    const apiAddrs = config.Addresses.API
 
+    this._apiServer = await Promise.resolve(
+      serverCreator.apply(this, [apiAddrs, this._createApiServer, hapiInfoToMultiaddr, ipfs])
```

Comment:
Why `await Promise.resolve` and `serverCreator.apply`? You can just call this function and await it:

```js
this._apiServers = await serverCreator(apiAddrs, this._createApiServer, ipfs)
```